src/runtime/propensity-server.js

```diff
@@ -29,18 +28,20 @@ export class PropensityServer {
    * Page configuration is known when Propensity API
    * is available, publication ID is therefore used
    * in constructor for the server interface.
+   * @param {!Window} win
    * @param {string} publicationId
    * @param {!../api/client-event-manager-api.ClientEventManagerApi} eventManager
+   * @param {!./fetcher.Fetcher} fetcher
    */
-  constructor(win, publicationId, eventManager) {
+  constructor(win, publicationId, eventManager, fetcher) {
     /** @private @const {!Window} */
     this.win_ = win;
     /** @private @const {string} */
     this.publicationId_ = publicationId;
     /** @private {?string} */
     this.clientId_ = null;
-    /** @private @const {!Xhr} */
-    this.xhr_ = new Xhr(win);
+    /** @private @const {!./fetcher.Fetcher} */
```

Comment:
I didn't realizer fetcher had a Fetcher. I thought fetcher was a Fetcher.

Reply:
Fetcher is a class XhrFetcher is an implementation of it.  I added `fetch()` to the class to match the one in AMP